logger/hooks/hooks.go

```diff
@@ -0,0 +1,42 @@
+// This file is part of Graylog.
+//
+// Graylog is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Graylog is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+
+package hooks
+
+import (
+	"path/filepath"
+
+	"github.com/rifflock/lfshook"
+	"github.com/Sirupsen/logrus"
+
+	"github.com/Graylog2/collector-sidecar/context"
+	"github.com/Graylog2/collector-sidecar/logger"
+)
+
+func AddLogHooks(context *context.Ctx, log *logrus.Logger) {
+	filesystemHook(context, log)
+}
+
+func filesystemHook(context *context.Ctx, log *logrus.Logger) {
+	logfile := filepath.Join(context.UserConfig.LogPath, "collector_sidecar.log")
+	writer := logger.GetRotatedLog(logfile, context.UserConfig.LogRotationTime, context.UserConfig.LogMaxAge)
```

Comment:
I am getting the following error when the logfile path does not exist:

```
Failed to fire hook: error: Failed to open file /dev/shm/sidecar/logs/collector_sidecar.log.%Y%m%d%H%M: open /dev/shm/sidecar/logs/collector_sidecar.log.201612220100: no such file or directory
INFO[0000] Starting signal distributor                  
Failed to fire hook: error: Failed to open file /dev/shm/sidecar/logs/collector_sidecar.log.%Y%m%d%H%M: open /dev/shm/sidecar/logs/collector_sidecar.log.201612220100: no such file or directory
INFO[0000] [filebeat] Starting (exec driver)            
Failed to fire hook: error: Failed to open file /dev/shm/sidecar/logs/collector_sidecar.log.%Y%m%d%H%M: open /dev/shm/sidecar/logs/collector_sidecar.log.201612220100: no such file or directory
```

How about creating the log file path before trying to write the logs?